test/e2e/plugins_test.go

```diff
@@ -0,0 +1,210 @@
+	TestPluginCode string = `#!/bin/bash
+
+echo "Hello Knative, I'm a Kn plugin"
+echo "  My plugin file is $0"
+echo "  I recieved arguments: $1 $2 $3 $4"`
+
+	KnConfigDefault string = `plugins-dir: %s
+lookup-plugins: %s`
+
+	FileModeReadable   = 0644
+	FileModeReadWrite  = 0666
+	FileModeExecutable = 0777
+)
+
+func TestPluginWorkflow(t *testing.T) {
+	var (
+		knConfigDir, knPluginsDir, knPluginsDir2  string
+		knConfigPath, knPluginPath, knPluginPath2 string
+		lookupPlugins                             bool
+		err                                       error
+	)
+
+	t.Parallel()
+	test := NewE2eTest(t)
+	setup := func(t *testing.T) {
+		test.createNamespaceOnSetup = false
+		test.Setup(t)
+		defer test.Teardown(t)
+
+		knConfigDir, err = ioutil.TempDir("", "kn-config")
+		assert.Assert(t, err == nil)
+
+		knPluginsDir = filepath.Join(knConfigDir, "plugins")
+		err = os.MkdirAll(knPluginsDir, FileModeExecutable)
+		assert.Assert(t, err == nil)
+
+		knPluginsDir2 = filepath.Join(knConfigDir, "plugins2")
+		err = os.MkdirAll(knPluginsDir2, FileModeExecutable)
+		assert.Assert(t, err == nil)
+
+		knConfigPath, err = test.createConfig(t, "config.yaml", "", knConfigDir)
+		assert.Assert(t, err == nil)
+		assert.Assert(t, knConfigPath != "")
+
+		knPluginPath, err = test.createPlugin(t, "kn-helloe2e", TestPluginCode, knPluginsDir)
+		assert.Assert(t, err == nil)
+		assert.Assert(t, knPluginPath != "")
+
+		knPluginPath2, err = test.createPlugin(t, "kn-hello2e2e", TestPluginCode, knPluginsDir2)
+		assert.Assert(t, err == nil)
+		assert.Assert(t, knPluginPath2 != "")
+	}
+
+	teardown := func(t *testing.T) {
+		err = os.RemoveAll(knConfigDir)
+		assert.Assert(t, err == nil)
+	}
+
+	t.Run("when kn config is empty", func(t *testing.T) {
+		t.Run("when using --plugin-dir", func(t *testing.T) {
+			t.Run("when --plugins-dir has a plugin", func(t *testing.T) {
+				t.Run("when --lookup-plugins is false", func(t *testing.T) {
+					lookupPlugins = false
+
+					setup(t)
+					defer teardown(t)
+
+					knFlags := []string{fmt.Sprintf("--plugins-dir=%s", knPluginsDir), "--lookup-plugins=false"}
+					assert.Assert(t, lookupPlugins == false)
+
+					t.Run("list plugin in --plugins-dir", func(t *testing.T) {
+						test.listPlugin(t, knFlags, []string{knPluginPath}, []string{})
+					})
+
+					t.Run("execute plugin in --plugins-dir", func(t *testing.T) {
+						test.runPlugin(t, knFlags, "helloe2e", []string{"e2e", "test"}, []string{"Hello Knative, I'm a Kn plugin", "I recieved arguments: e2e"})
+					})
+
+					t.Run("does not list any other plugin in $PATH", func(t *testing.T) {
+						test.listPlugin(t, knFlags, []string{knPluginPath}, []string{knPluginPath2})
+					})
+				})
+
+				t.Run("with --lookup-plugins is true", func(t *testing.T) {
+					lookupPlugins = true
+
+					setup(t)
+					defer teardown(t)
+
+					knFlags := []string{fmt.Sprintf("--plugins-dir=%s", knPluginsDir), "--lookup-plugins=true"}
+					assert.Assert(t, lookupPlugins == true)
```

Comment:
```suggestion

```

unnecessary check

Reply:
Well because the `lookupPlugins` is a variable, the `setup(t)` could have changed it and since this block is about `lookupPlugins` either `true` or `false` wanted to ensure that.